src/OpenCensus/Utils/Arguments.cs

```diff
@@ -0,0 +1,42 @@
+﻿// <copyright file="Arguments.cs" company="OpenCensus Authors">
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of theLicense at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenCensus.Utils
+{
+    using System;
+
+    /// <summary>
+    /// Method for arguments validation.
+    /// </summary>
+    public class Arguments
+    {
+        /// <summary>
+        /// Validates a condition.
+        /// </summary>
+        /// <param name="condition">Boolean condition</param>
+        /// <param name="errorMessage">Error that is thrown in case of condition being false</param>
+        /// <returns>Whether condition is true</returns>
+        public static bool Check(bool condition, string errorMessage)
+        {
+            if (!condition)
```

Comment:
this is one-liner in code. Do you need a separate method defined?

Reply:
What other overloads are you anticipating? From my expirience these helper methods only complicates understanding the code. You never know will it throw or ignore an error for instance.

Reply:
As far as I understand, this is a simplified guava preconditions implementation is C#. It would be cool to have something like that on the platform or common-lib level, but unfortunately, there is nothing like that and couple 3rd party I've found - [CuttingEdge.Conditions](https://www.nuget.org/packages/CuttingEdge.Conditions) and [Preconditions.NET](https://www.nuget.org/packages/Preconditions.NET) are not really popular.

We also don't get any static code analysis from this when we implement them on our own. 

The problem here is not only readability but also perf. A caller has to create an error string (which is not const) and sometimes do a bunch of operations to make error message descriptive enough (add values of Ids, some context).
And this string will be constructed even when the condition is false (almost always). So such checks should not be used on the hot path anyways. 

So I tend to agree with Sergey, there are certain issues with this approach and I'd prefer we explicitly throw or write log in the code where it's needed.

Reply:
OK, I'm removing this in the meantime.